aries/aries_vcx/src/common/signing.rs

```diff
@@ -8,6 +8,19 @@ use time;
 
 use crate::errors::error::prelude::*;
 
+// Utility function to handle both padded and unpadded Base64URL data
+fn base64url_decode(encoded: &str) -> VcxResult<Vec<u8>> {
+    general_purpose::URL_SAFE_NO_PAD
+        .decode(encoded)
+        .or_else(|_| general_purpose::URL_SAFE.decode(encoded))
+        .map_err(|err| {
+            AriesVcxError::from_msg(
+                AriesVcxErrorKind::InvalidJson,
+                format!("Cannot decode Base64URL data: {:?}", err),
+            )
+        })
```

Comment:
nice, this should definitely work. However as a nitpick, it would be preferred to intentionally trigger the error path in order to try `general_purpose::URL_SAFE.decode`.

In other languages, this would sort of be like wrapping it with a `try catch` block, and trying again in the `catch` block. this is usually an anti-pattern because throwing exceptions has cost. In Rust there are no exceptions, just Errors, and Errors are less costly... however IMO it would still be nicer to avoid this pattern.

Additionally, I'm only making this comment because I've had experience with this situation when using the `base64` crate before and found that there is a better way. `URL_SAFE_NO_PAD` and `URL_SAFE` are pre-made instances of `GeneralPurpose` base64 engines. I'd have a look into these implementations and see if you can construct your own `GeneralPurpose` engines which use the URL_SAFE alphabet, but knows how to handle with or without padding (i.e. it is lenient/indifferent to padding when decoding). 

Once you construct one of those, i'd make it into a `const URL_SAFE_<some appropriate name>: GeneralPurpose` like how they do with `URL_SAFE_NO_PAD` and `URL_SAFE`.

Reply:
oh also, if you could, maybe add a comment above the const explaining what it is, and maybe explain that in the future this const engine could be used globally in the crate:
```
NOTE: we may wish to move and use this engine as a global util in the aries-vcx project
```

Also note, there is an example of all of this within the aries-vcx repo... i forgot where..